cabal-install/Distribution/Client/ProjectBuilding.hs

```diff
@@ -1133,10 +1133,11 @@ buildInplaceUnpackedPackage verbosity
       | elabBuildHaddocks pkg = action
       | otherwise            = return ()
 
-    whenReRegister  action = case buildStatus of
-      BuildStatusConfigure          _ -> action
-      BuildStatusBuild Nothing      _ -> action
-      BuildStatusBuild (Just mipkg) _ -> return mipkg
+    whenReRegister  action
+      = case buildStatus of
+          BuildStatusBuild (Just mipkg) _ -> return mipkg
+          _ | null (elabBuildTargets pkg) -> return Nothing -- ick
```

Comment:
Well, I half wanted to just drop the recorded InstalledPackageInfo entirely; I couldn't figure out where it was being used. Adjusting it to have another case certainly seems like it could work; I wonder if there is something simpler.